crates/katana/primitives/src/genesis/json.rs

```diff
@@ -234,8 +244,15 @@ impl GenesisJson {
         path.pop();
 
         let mut genesis: Self = serde_json::from_reader(BufReader::new(file))?;
+        let mut class_name_map = HashMap::new();
+         for class in &genesis.classes {
+            if let Some(name) = &class.name {
+                class_name_map.insert(name.to_string(), class.class_hash.unwrap());  
+            }
+         }
+
         // resolves the class paths, if any
-        genesis.resolve_class_artifacts(path)?;
+        genesis.resolve_class_artifacts(path, &class_name_map)?;
```

Comment:
```suggestion
        genesis.resolve_class_artifacts(path)?;
```

This functions resolves the artifacts, which consist on loading the class JSON into memory to be available for genesis operation. The name here shouldn't be relevant as we are expecting to use the names into other entries, not the classes themselves.